com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensor.cs

```diff
@@ -4,28 +4,96 @@
 
 namespace MLAgents
 {
-    public class RayPerceptionSensor : ISensor
+    public enum RayPerceptionCastType
+    {
+        Cast2D,
+        Cast3D,
+    }
+
+    public class RayPerceptionInput
     {
-        public enum CastType
+        // TODO
+        /// <param name="unscaledRayLength"></param>
+        /// <param name="rayAngles">List of angles (in degrees) used to define the rays. 90 degrees is considered
+        ///     "forward" relative to the game object</param>
+        /// <param name="detectableObjects">List of tags which correspond to object types agent can see</param>
+        /// <param name="startOffset">Starting height offset of ray from center of agent.</param>
+        /// <param name="endOffset">Ending height offset of ray from center of agent.</param>
+        /// <param name="unscaledCastRadius">Radius of the sphere to use for spherecasting. If 0 or less, rays are used
+        /// instead - this may be faster, especially for complex environments.</param>
+        /// <param name="transform">Transform of the GameObject</param>
+        /// <param name="castType">Whether to perform the casts in 2D or 3D.</param>
+        /// <param name="perceptionBuffer">Output array of floats. Must be (num rays) * (num tags + 2) in size.</param>
+        /// <param name="layerMask">Filtering options for the casts</param>
+        /// <param name="debugInfo">Optional debug information output, only used by RayPerceptionSensor.</param>
+
+        public float rayLength;
```

Comment:
TODO add transform as a field here too, remove from PerceiveStatic args

Reply:
transform for scale, etc.?

Reply:
Transform of the game object (usually the Agent's object, but I guess it could be something else in theory).